### PR TITLE
[10.0]stock_available: improve computation performance

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
Forward port of https://github.com/OCA/stock-logistics-warehouse/pull/249

### CODE REVIEW

File: stock_available/models/product_template.py [hunk 1]
```diff
@@ -10,6 +10,33 @@
...
+        for product in self:
+
+            if isinstance(product.id, models.NewId):
+                continue
+            immediately_usable_qty = 0.0
+
+            for p in product.product_variant_ids:
+                qty = variant_available[p.id]["immediately_usable_qty"]
+                immediately_usable_qty += qty
+            if product.product_variant_ids:
```
Comment (rousseldenis) [inline stock_available/models/product_template.py]: @Cedric-Pigeon Can you maybe put this computation in the above loop?

File: stock_available/models/product_template.py [hunk 2]
```diff
@@ -10,6 +10,33 @@
...
+                potential_qty = max(
+                    [v.potential_qty for v in product.product_variant_ids])
+            res[product.id].update({
```
Comment (rousseldenis) [inline stock_available/models/product_template.py]: @Cedric-Pigeon One update ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by Cedric-Pigeon** [comment]
@rousseldenis Remarks integrated, thanks.

**Review by rousseldenis — APPROVED** [review]
:+1:

**Review by lmignon — APPROVED** [review]
:+1: Thank you @Cedric-Pigeon

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza What about this one ?

**Comment by lmignon** [comment]
@lasley can you take a look at this one. It's an important improvement. The pr is falsely red due to a decrease in the coverage since the fix implies more lines of code.

**Comment by pedrobaeza** [comment]
Well, the patch seems legit, but unfortunately, there's not automated test to confirm that it works correctly at least one time. That's why we shouldn't look down on that check. But if the PSC decides to merge it, it doesn't go against OCA rules, as tests are not (yet) mandatory.

**Comment by Cedric-Pigeon** [comment]
@pedrobaeza You are right. We just provided a test.

**Pull request merged**